torch/nn/modules/transformer.py

```diff
@@ -233,12 +233,13 @@ class TransformerEncoderLayer(Module):
         nhead: the number of heads in the multiheadattention models (required).
         dim_feedforward: the dimension of the feedforward network model (default=2048).
         dropout: the dropout value (default=0.1).
+        activation: the activation function of intermediate layer, relu or gelu (default=relu).
 
     Examples::
         >>> encoder_layer = nn.TransformerEncoderLayer(d_model, nhead)
     """
 
-    def __init__(self, d_model, nhead, dim_feedforward=2048, dropout=0.1):
+    def __init__(self, d_model, nhead, dim_feedforward=2048, dropout=0.1, activation="relu"):
```

Comment:
The current changes are on the encoder/decoder layer level. It won't affect Transformer/TransformerEncoder/TransformerDecoder and users still need to build their own. Is this what you want? If you can show/explain me your application, we can briefly discuss the design for this case.

Reply:
I was using ``TransformerEncoderLayer`` to build bert model with pretrained weights.
```
encoder_layer = nn.TransformerEncoderLayer(d_model, nhead, dim_feedforward=4*d_model, activation="gelu")
transformer_encoder = nn.TransformerEncoder(encoder_layer, num_layers)
```

Then I found that the output of the encoder layer is not right. So I dig into it and found the difference is `relu` and `gelu`

Reply:
Also, it may be useful for `Transformer/TransformerEncoder/TransformerDecoder` to have `activation` as kwargs. If you agree, I could add it.

Reply:
I think it makes some senses. I see some applications have the activation fn args in Transformer/TransformerEncoder/TransformerDecoder. IMO, we can add the option for TransformerEncoder/TransformerDecoder. In that case, you can use nn.TransformerEncoder directly.

Reply:
Yes, it makes sense.

Reply:
I added `activation` in `Transformer`. `TransformerEncoder/TransformerDecoder` accept `encoder_layer` as argument, so there is no need to add it. Please review the following commits.